specification/quota/resource-manager/Microsoft.Quota/preview/2021-03-15-preview/quota.json

```diff
@@ -0,0 +1,1252 @@
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ExceptionResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "QuotaInformation"
+        ],
+        "description": "Create or update the quota limit for the specified resource to the requested value. To update the quota, follow these steps:\n1. Use the GET operation to determine how much quota remains for the specific resource and to calculate the new quota limit. These steps are detailed in [this example](https://techcommunity.microsoft.com/t5/azure-governance-and-management/using-the-new-quota-rest-api/ba-p/2183670).\n2. Use this PUT operation to update the quota limit.",
+        "operationId": "Quota_CreateOrUpdate",
+        "x-ms-examples": {
+          "Quotas_Request_ForCompute": {
+            "$ref": "./examples/putComputeOneSkuQuotaRequest.json"
+          },
+          "Quotas_Request_ForMachineLearningServices_DedicatedResource": {
+            "$ref": "./examples/putMachineLearningServicesQuotaRequestDedicated.json"
+          },
+          "Quotas_Request_ForMachineLearningServices_LowPriorityResource": {
+            "$ref": "./examples/putMachineLearningServicesQuotaRequestLowPriority.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceNameInParameters"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ScopeInPath"
+          },
+          {
+            "name": "createQuotaRequest",
+            "in": "body",
+            "description": "Quota requests payload.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CurrentQuotaLimitBase"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Returns the quota request details.",
+            "schema": {
+              "$ref": "#/definitions/QuotaRequestOneResourceSubmitResponse"
+            }
+          },
+          "201": {
+            "description": "The request is accepted and is being processed. To check the status for this request, use the value for the quota request ID **id** obtained in the response of a [Quota Request Status](https://docs.microsoft.com/en-us/rest/api/reserved-vm-instances/quotarequeststatus/get) GET operation.",
+            "schema": {
+              "$ref": "#/definitions/QuotaRequestSubmitResponse202"
```

Comment:
>QuotaRequestSubmitResponse202 [](http://example.com/codeflow?start=37&length=29)

It should be CurrentQuotaLimitBase. PUT should return what GET returns not async operation status response.

[https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc008-put-get-patch--list-must-return-the-same-resource-schema](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc008-put-get-patch--list-must-return-the-same-resource-schema)

Reply:
Updated to CurrentQuotaLimitBase